drivers/virtio/net/virtio_net.c

```diff
@@ -330,7 +334,13 @@ static int virtio_netdev_rx_fillup(struct virtio_net_device *vndev,
 				status |= UK_NETDEV_STATUS_UNDERRUN;
 				goto out;
 			}
-			filled += 2;
+			/* This count is not necessarily correct in the
+			 * presence of paging, but should be close enough as an
+			 * approximation.
+			 * TODO: Use indirect descriptors to ensure we are
```

Comment:
We should probably also add an assertion here?

Reply:
No, as the comment hints, having too many buffers is not fatal. Also the problem is that we don't have an precise number before trying to add the buffer. So adding an assertion would just cause random panics.